crates/typst-library/src/visualize/color.rs

```diff
@@ -210,30 +234,31 @@ pub enum Color {
     Hsv(Hsv),
 }
 
-#[scope]
-impl Color {
+scope_with_color_definitions! {
+    Color,
+
+    BLACK: Luma -> 0.0, 1.0,
+    GRAY: Luma -> 0.6666666, 1.0,
+    WHITE: Luma -> 1.0, 1.0,
+    SILVER: Luma -> 0.8666667, 1.0,
+    NAVY: Rgb -> 0.0, 0.121569, 0.247059, 1.0,
+    BLUE: Rgb -> 0.0, 0.454902, 0.85098, 1.0,
+    AQUA: Rgb -> 0.4980392, 0.858823, 1.0, 1.0,
+    TEAL: Rgb -> 0.223529, 0.8, 0.8, 1.0,
+    EASTERN: Rgb -> 0.13725, 0.615686, 0.678431, 1.0,
+    PURPLE: Rgb -> 0.694118, 0.050980, 0.788235, 1.0,
+    FUCHSIA: Rgb -> 0.941177, 0.070588, 0.745098, 1.0,
+    MAROON: Rgb -> 0.521569, 0.078431, 0.294118, 1.0,
+    RED: Rgb -> 1.0, 0.254902, 0.211765, 1.0,
+    ORANGE: Rgb -> 1.0, 0.521569, 0.105882, 1.0,
+    YELLOW: Rgb -> 1.0, 0.8627451, 0.0, 1.0,
+    OLIVE: Rgb -> 0.239216, 0.6, 0.4392157, 1.0,
+    GREEN: Rgb -> 0.1803922, 0.8, 0.2509804, 1.0,
+    LIME: Rgb -> 0.0039216, 1.0, 0.4392157, 1.0;
+
     /// The module of preset color maps.
     pub const MAP: fn() -> Module = || typst_utils::singleton!(Module, map()).clone();
```

Comment:
I wonder if we could get `#[scope]` to work across multiple impl blocks so we can split this into  `scope_with_color_definitions! { Color, ... }` and `#[scope] impl Color { ... }` for more flexibility (would allow multiple such macros).

But that seems complex enough to warrant a separate PR, and in the worst case scenario could require linking hacks[^1] to work properly.

[^1]: https://docs.rs/inventory/